read.c

```diff
@@ -590,8 +590,11 @@ int redisReaderGetReply(redisReader *r, void **reply) {
 
     /* Emit a reply when there is one. */
     if (r->ridx == -1) {
-        if (reply != NULL)
+        if (reply != NULL) {
             *reply = r->reply;
+        } else if (r->reply != NULL && r->fn && r->fn->freeObject) {
+            r->fn->freeObject(r->reply);
+        }
```

Comment:
This is a dead branch, program will never go into here.

Reply:
If I call the function like redisReaderGetReply(r, NULL); ?  Is it a dead branch?

Reply:
@movebean I see the difference. you're right.